### PR TITLE
Adjust diff granularity

### PR DESCRIPTION
**Pull request by Denyer**
**Changes**

- Updates the diff in the following ways:
  - Remove upper limit on diff
  - Split changes across nodes
  - Split insertions that span multiple nodes

**Notes for reviewers**

- Easiest to go commit by commit, the commit messages should suffice in explaining the change alongside the accompanying test
- Use yarn link to and build this locally, try checking the diff on multiple documents

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Run Tests
```
Comment (dakotalraymond) [inline .github/workflows/run_tests.yml:1]: 🙌

File: src/diff.js [hunk 2]
```diff
@@ -110,6 +105,47 @@ function findFencedNodes(a) {
 // getting in the way of the diff algorithm when comparing characters.
 const FORCED_MATCH_CHAR = '@'
 
+const NODES_NAMES = [
```
Comment (dakotalraymond) [inline src/diff.js]: I think this name could be a little bit more descriptive. Maybe something like `BOUNDARY_NODES`?

**Review by dakotalraymond — APPROVED** [review]
From everything I can see, this looks good. I say we get it shipped so we can get more eyes on it, and then add subsequent tests for any issues we find.

File: src/diff.js (hunk 2 above)
Comment (Denyer) [reply src/diff.js]: Yeah agree, the name is not clear

**Comment by Denyer** [comment]
@dakotalraymond thanks for the review, there is one failing test which I've struggled to follow, it might be that the expected value just needs to be updated, but would be great to get your help with it

**Pull request merged**